akka-actor-typed-tests/src/test/scala/akka/actor/typed/TerminatedSpec.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed
+
+import akka.actor.testkit.typed.scaladsl.TestInbox
+import org.scalatest.{ Matchers, WordSpec }
+
+class TerminatedSpec extends WordSpec with Matchers {
+
+  "Child Failed" must {
+    "should be pattern matchable" in {
+
+      val probe = TestInbox[String]()
+      val ex = new RuntimeException("oh dear")
+      val childFailed = new ChildFailed(probe.ref, ex)
```

Comment:
I like all the metadata available in this but is it perhaps too heavy weight?

Reply:
I guess I mean do we need the ref or just the ActorPath. I'm sure users would want the Throwable.

Reply:
That said, I can definitely see use cases where you'd want to have access to that `ref` in question.

Reply:
Yep you get the ref for the regular Terminated case and a new ActorRef can have the same ActorPath as an old ActorRef